### ISSUE TITLE
Add `ComponentLike` argument to `#failWithAdventureComponent` method

### ISSUE THREAD

**Issue by ipiepiepie** [JorelAli/CommandAPI#577]
### Description

I'd like to send error messages, using my custom message class, which implements `ComponentLike` interface.

At the moment, my error handling in commands looks like this:
```Java
if (!player.hasTown())
    throw CommandAPIBukkit.failWithAdventureComponent(message(ERR_WITHOUT_TOWN).asComponent());
```
After implementing suggested feature, my error handling would be:
```Java
if (!player.hasTown())
    throw CommandAPIBukkit.failWithAdventureComponent(message(ERR_WITHOUT_TOWN));
```

The second option looks much easier and prettier.

### Expected code

```Java
CommandAPIBukkit#failWithAdventureComponent(ComponentLike)
```

**Comment by Strokkur424** [comment]
Seems like it might be a quick and easy feature to add, without any drawbacks. Good idea, I approve

**Comment by DerEchtePilz** [comment]
Implemented in version 9.5.2
